tools/windows/install-help/cal/caninstall.cpp

```diff
@@ -1,6 +1,86 @@
 #include "stdafx.h"
 #include "TargetMachine.h"
 
+bool canInstall(const CustomActionData &data, bool &bResetPassword, std::wstring *resultMessage)
+{
+    bool ddUserExists = false;
+    bool ddServiceExists = false;
+    bool isDC = false;
+    bool isReadOnlyDC = false;
+    bool isNtAuthority = false;
+    bool isServiceAccount = false;
+    bool isUserDomainUser = false;
+    bool haveUserPassword = false;
+    std::wstring userDomain;
+    std::wstring computerDomain;
+
+    // ddUserExists
+    ddUserExists = data.DoesUserExist();
+
+    // isServiceAccount
+    isServiceAccount = data.IsServiceAccount();
+
+    // ddServiceExists
+    {
+        auto result = doesServiceExist(agentService);
+        if (-1 == result) {
+            // error, call it false?
+            ddServiceExists = false;
+        } else if (0 == result) {
+            ddServiceExists = false;
+        } else if (1 == result) {
+            ddServiceExists = true;
+        }
+    }
+
+    // isNtAuthority
+    if (ddUserExists) {
+        const auto ntAuthoritySid = WellKnownSID::NTAuthority();
+        if (!ntAuthoritySid.has_value())
+        {
+            WcaLog(LOGMSG_STANDARD, "Cannot check user SID against NT AUTHORITY: memory allocation failed");
+        }
+        else
+        {
+            isNtAuthority = EqualPrefixSid(data.Sid(), ntAuthoritySid.value().get());
+        }
+    } else {
+        isNtAuthority = false;
+    }
```

Comment:
mixing figure bracket style. Not a big deal just wondering why. I think I have seen it in other place